tests/e2e_tests/flatfield_e2e.py

```diff
@@ -0,0 +1,368 @@
+    l1_flatfield_filelist.sort()
+    
+    # define the raw science data to process
+
+    mock_cal_filelist = l1_dark_filelist[-2:]
+
+    ###### Setup necessary calibration files
+    # Create necessary calibration files
+    # we are going to make calibration files using
+    # a combination of the II&T nonlinearty file and the mock headers from
+    # our unit test version
+    pri_hdr, ext_hdr = mocks.create_default_headers()
+    ext_hdr["DRPCTIME"] = time.Time.now().isot
+    ext_hdr['DRPVERSN'] =  corgidrp.__version__
+    mock_input_dataset = data.Dataset(mock_cal_filelist)
+
+    this_caldb = caldb.CalDB() # connection to cal DB
+
+    # Nonlinearity calibration
+    nonlin_dat = np.genfromtxt(nonlin_path, delimiter=",")
+    nonlinear_cal = data.NonLinearityCalibration(nonlin_dat, pri_hdr=pri_hdr, ext_hdr=ext_hdr,
+                                                input_dataset=mock_input_dataset)
+    nonlinear_cal.save(filedir=flat_outputdir, filename="mock_nonlinearcal.fits" )
+    this_caldb.create_entry(nonlinear_cal)
+
+    # KGain
+    kgain_val = 8.7
+    kgain = data.KGain(np.array([[kgain_val]]), pri_hdr=pri_hdr, ext_hdr=ext_hdr, 
+                    input_dataset=mock_input_dataset)
+    kgain.save(filedir=flat_outputdir, filename="mock_kgain.fits")
+    this_caldb.create_entry(kgain)
+
+    # NoiseMap
+    with fits.open(fpn_path) as hdulist:
+        fpn_dat = hdulist[0].data
+    with fits.open(cic_path) as hdulist:
+        cic_dat = hdulist[0].data
+    with fits.open(dark_path) as hdulist:
+        dark_dat = hdulist[0].data
+    noise_map_dat_img = np.array([fpn_dat, cic_dat, dark_dat])
+    noise_map_dat = np.zeros((3, detector.detector_areas['SCI']['frame_rows'], detector.detector_areas['SCI']['frame_cols']))
+    rows, cols, r0c0 = detector.unpack_geom('SCI', 'image')
+    noise_map_dat[:, r0c0[0]:r0c0[0]+rows, r0c0[1]:r0c0[1]+cols] = noise_map_dat_img
+    noise_map_noise = np.zeros([1,] + list(noise_map_dat.shape))
+    noise_map_dq = np.zeros(noise_map_dat.shape, dtype=int)
+    err_hdr = fits.Header()
+    err_hdr['BUNIT'] = 'detected EM electrons'
+    ext_hdr['B_O'] = 0
+    ext_hdr['B_O_ERR'] = 0
+    noise_map = data.DetectorNoiseMaps(noise_map_dat, pri_hdr=pri_hdr, ext_hdr=ext_hdr,
+                                    input_dataset=mock_input_dataset, err=noise_map_noise,
+                                    dq = noise_map_dq, err_hdr=err_hdr)
+    noise_map.save(filedir=flat_outputdir, filename="mock_detnoisemaps.fits")
+    this_caldb.create_entry(noise_map)
+
+    ####### Run the walker on some test_data
+
+    recipe = walker.autogen_recipe(l1_flatfield_filelist, flat_outputdir)
+     ### Modify they keywords of some of the steps
+    for step in recipe['steps']:
+        # if step['name'] in ["desmear", "cti_correction"]:
+        #     step['skip'] = True
+        if step['name'] == "create_onsky_flatfield":
+            step['keywords']['n_pix'] = 165 # full shaped pupil FOV
```

Comment:
why do we have to modify this here?

Reply:
because we're testing full FOV?

Reply:
We don't have too. I just wanted to leave this here, because this value may be the first thing we tune, depending on whether we are doing the full shaped pupil FOV or the smaller HLC FOV>